addons/mrp/models/mrp_production.py

```diff
@@ -2895,9 +2895,11 @@ def _update_order_line_info(self, product_id, quantity, child_field=False, **kwa
             else:
                 entity.unlink()
         elif quantity > 0:
-            new_line = self._get_new_catalog_line_values(product_id, quantity, **kwargs)
-            command = Command.create(new_line)
+            new_line_vals = self._get_new_catalog_line_values(product_id, quantity, **kwargs)
+            command = Command.create(new_line_vals)
             self.write({child_field: [command]})
+            new_line = self[child_field].filtered(lambda mv: mv.product_id.id  == product_id)[-1:]
+            self._update_catalog_line_quantity(new_line, quantity, **kwargs)
```

Comment:
To me, new moves should be marked as `additional` to be confirmed in `autoconfirm_production`
 https://github.com/odoo/odoo/blob/f8567090718faaa0cbef9a2ca3ab2e48b1ac2088/addons/mrp/models/mrp_production.py#L1307